### PR TITLE
add renderMarkdown to ICompiler

### PR DESCRIPTION
**Pull request by giladgray**
KssPlugin markdownifies documentation + markup for syntax highlighting

### CODE REVIEW

**Comment by giladgray** [comment]
@adidahiya mind reviewing this as @themadcreator is out today?

**Comment by giladgray** [comment]
though i suppose it could wait till monday

File: src/plugins/kss.ts [hunk 1]
```diff
@@ -57,18 +65,19 @@ export class KssPlugin implements IPlugin<IKssPluginData> {
     }
 }
 
-function convertSection(section: kss.ISection): IKssExample {
+function convertSection(this: ICompiler, section: kss.ISection): IKssExample {
```
Comment (themadcreator) [inline src/plugins/kss.ts]: using map's `this` context in this fashion is pretty hacky. I know it's more performant than using a lambda in the map above, but just one more arg and this doesn't work.

I think i'd prefer to just use an explicit argument and a lambda in the map.

File: src/plugins/kss.ts [hunk 2]
```diff
@@ -57,18 +65,19 @@ export class KssPlugin implements IPlugin<IKssPluginData> {
...
     return {
-        documentation: section.description(),
+        documentation: this.renderMarkdown(section.description()),
         markup: section.markup() || "",
-        modifiers: section.modifiers().map(convertModifier),
+        markupHtml: this.renderMarkdown(`\`\`\`html\n${section.markup() || ""}\n\`\`\``),
+        modifiers: section.modifiers().map(convertModifier, this),
```
Comment (themadcreator) [inline src/plugins/kss.ts]: And again =(

File: src/plugins/plugin.ts [hunk 3]
```diff
@@ -58,6 +58,11 @@ export interface ICompiler {
      * is.
      */
     renderBlock: (blockContent: string, reservedTagWords?: string[]) => IBlock;
+
+    /**
+     * Render a string of markdown to HTML, using the options from `Documentalist`.
+     */
+    renderMarkdown: (markdown: string) => string;
```
Comment (themadcreator) [inline src/plugins/plugin.ts]: 👍

File: src/plugins/kss.ts (hunk 1 above)
Comment (giladgray) [inline src/plugins/kss.ts]: but it's _so damn clever_ :woman_scientist:

Comment (themadcreator) [inline src/plugins/kss.ts]: I know, and kudos. But simple and clear is better, IMO

**Pull request merged**